src/core/src/test/java/org/apache/jmeter/gui/util/JSyntaxTextAreaTest.java

```diff
@@ -41,7 +45,37 @@ public void testSetLanguage() {
             textArea.setLanguage(null);
             assertEquals(SyntaxConstants.SYNTAX_STYLE_NONE, textArea.getSyntaxEditingStyle());
         } catch (HeadlessException he) {
-            // Does not work in headless mode
+            // Does not work in headless mode, which depends on value of java.awt.headless property
+            // and the OS (e.g. might work on MacOS and not on Linux due to missing X11).
+            System.out.println("WARNING for JSyntaxTextAreaTest.testSetLanguage test: does not work in headless mode");
+        }
+    }
+
+    @Test
+    public void testHeadless() {
+        String key = "java.awt.headless";
+        String initialValue = System.getProperty(key);
+        try {
+            System.setProperty(key, "true");
+            // getInstance returns anonymous class with some overridden methods
+            // to avoid errors due to 'java.awt.headless=true'
+            JSyntaxTextArea textArea = JSyntaxTextArea.getInstance(10,20);
+
+            String myText = "my text";
+            textArea.setText(myText);
```

Comment:
Are you testing set/getText here, or the creation of a specialised subclass?

Reply:
Test if the textArea returned by getInstance() in headless mode is retaining the text. Without the fix of the specialised subclass this will fail, because getText() returns null.

Reply:
Right, but why the call ` textArea.setCodeFoldingEnabled(true);`? It seems that you want to test, that we get a special variant, that works in headless mode. But in my eyes, you know, that it works in headless mode (or could not check it anyway) and this test is about saving the text. One test should assert one thing (at least new ones should :))

Reply:
agree, I removed the NPE implicit check